test/compression.js

```diff
@@ -661,6 +699,70 @@ function createServer (opts, fn) {
   })
 }
 
+function createHttp2Server (opts, fn) {
+  var _compression = compression(opts)
+  var server = http2.createServer(function (req, res) {
+    req.on('error', function (error) {
+      console.error('An error event occurred on a http2 request.', error)
+    })
+    res.on('error', function (error) {
+      console.error('An error event occurred on a http2 response.', error)
+    })
+    _compression(req, res, function (err) {
+      if (err) {
+        res.statusCode = err.status || 500
+        res.end(err.message)
+        return
+      }
+
+      fn(req, res)
+    })
+  })
+  server.on('error', function (error) {
+    console.error('An error event occurred on the http2 server.', error)
+  })
+  server.on('sessionError', function (error) {
+    console.error('A sessionError event occurred on the http2 server.', error)
+  })
+  server.listen(0, '127.0.0.1')
+  return server
+}
+
+function createHttp2Client (port) {
+  var client = http2.connect('http://127.0.0.1:' + port)
+  client.on('error', function (error) {
+    console.error('An error event occurred in the http2 client stream.', error)
+  })
+  return client
+}
+
+function closeHttp2 (request, client, server, callback) {
+  if (typeof client.shutdown === 'function') {
+    // this is the node v8.x way of closing the connections
+    request.destroy(http2.constants.NGHTTP2_NO_ERROR, function () {
+      client.shutdown({}, function () {
+        server.close(function () {
+          callback()
+        })
+      })
+    })
+  } else {
+    // this is the node v9.x onwards way of closing the connections
+    request.close(http2.constants.NGHTTP2_NO_ERROR, function () {
+      client.close(function () {
+        // force existing connections to time out after 1ms.
+        // this is done to force the server to close in some cases where it wouldn't do it otherwise.
+        server.setTimeout(1, function () {
+          console.warn('An http2 connection timed out instead of being closed properly.')
```

Comment:
So the comment says "force existing connections to time out after 1ms.", but inside the timeout there is nothing but a console warn. Should there be something being don in here? Otherwise, what purpose does the timeout serve? If to fail the test, should an error be passed to callback instead to make the test fail?